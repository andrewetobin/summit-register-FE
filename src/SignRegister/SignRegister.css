.SignRegister {
  border: 1px solid black;
  border-radius: 20px;
  font-size: 16px;
  opacity: .6;
  padding: 5px 10px;
}

.active {
  background: rgb(1, 162, 1);
  color: white;
  opacity: 1;
  transition: all 0.4s ease-in-out;
}

.active:hover {
  background: white;
  color: rgb(1, 162, 1);
  cursor: pointer;
  opacity: 1;
  transition: all 0.4s ease-in-out;
}
