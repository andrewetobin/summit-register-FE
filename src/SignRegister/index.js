import React, { Component } from "react";
import PropTypes from "prop-types";
import "./SignRegister.css";

class SignRegister extends Component {
  render() {
    const { withinRange, handleSignLog } = this.props;
    let buttonText;

    withinRange
      ? (buttonText = "Sign Register")
      : (buttonText = "Out of Range");

    return (
      <button className={withinRange ? "SignRegister active" : "SignRegister"} onClick={() => handleSignLog()}>
        {buttonText}
      </button>
    );
  }
}

SignRegister.propTypes = {
  handleLogUpdate: PropTypes.func
};

export default SignRegister;
