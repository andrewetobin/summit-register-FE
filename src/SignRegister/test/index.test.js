import React from 'react';
import ReactDOM from 'react-dom';
import SignRegister from '..';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SignRegister />, div);
  ReactDOM.unmountComponentAtNode(div);
});