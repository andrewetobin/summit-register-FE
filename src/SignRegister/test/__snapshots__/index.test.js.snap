// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignRegister should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignRegister
    handleSignLog={[MockFunction]}
    withinRange={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Sign Register",
      "className": "SignRegister",
      "disabled": true,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Sign Register",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Sign Register",
        "className": "SignRegister",
        "disabled": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Sign Register",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
